src/app/docs/[mode]/[branch]/[[...path]]/@print/page.tsx

```diff
@@ -0,0 +1,149 @@
+import type { Metadata } from 'next';
+import matter from 'gray-matter';
+
+import React from 'react';
+import { ContentLoader, ContentPrint } from '@/components/Layouts';
+
+import { siteConfig } from '@/config';
+import { notFound } from 'next/navigation';
+import { getFileContent } from '@/lib/Github';
+import { getLogger } from '@/lib/Logger';
+import type { ContentItem, MatterData } from '@/lib/Types';
+const logger = getLogger().child({ namespace: 'docs/page' });
+logger.level = 'error';
+export const metadata: Metadata = {
+  title: 'Airview',
+  description: 'Airview AI',
+};
+
+async function checkFrontmatter(content: string, context: ContentItem) {
+  const matterData = matter(content, {
+    excerpt: false,
+  }).data as MatterData;
+  if (matterData) {
+    Object.keys(matterData).forEach((key) => {
+      if (matterData && matterData[key] && matterData[key] instanceof Date) {
+        matterData[key] = (matterData[key] as Date).toISOString();
+      }
+    });
+  }
+  if (
+    matterData &&
+    matterData.external_repo &&
+    matterData.external_owner &&
+    matterData.external_path &&
+    matterData.git_provider
+  ) {
+    const { external_repo, external_owner, external_path } = matterData;
+    const owner = external_owner as string;
+    const repo = external_repo as string;
+    const branch = context.branch;
+    const file = external_path as string;
+    let pageContent;
+    try {
+      pageContent = await getFileContent({ owner, repo, path: file });
+    } catch (error) {
+      logger.error({ msg: 'checkFrontmatter: ', error });
+    }
+    let linkedContent = '';
+    if (pageContent && pageContent.content) {
+      linkedContent = pageContent?.content
+        ? Buffer.from(pageContent.content).toString()
+        : '';
+    }
+    return {
+      content: linkedContent,
+      context: { ...context, linked: { owner, repo, branch, path: file } },
+    };
+  } else {
+    return { content, context };
+  }
+}
+
+export default async function Page({
+  params,
+}: {
+  params: { mode: 'view' | 'edit' | 'print'; branch: string; path: string[] };
+}) {
+  if (
+    params.path &&
+    params.path[0] &&
+    siteConfig.content[params.path[0] as keyof typeof siteConfig.content]
+  ) {
+    let path = params.path;
+
+    let file = '';
+
+    // if 'related_config' is somewhere in the path, then the file is the last element in the path
+    if (path.includes('related_content')) {
+      // join all the parts after related_config
+      file = path
+        .slice(path.indexOf('related_content') + 1)
+        .join('/') as string;
+    } else {
+      file = path.join('/') as string;
+    }
```

Comment:
```suggestion
    let filepath = '';

    // if 'related_config' is somewhere in the path, then the file is the last element in the path
    if (path.includes('related_content')) {
      // join all the parts after related_config
      file = path
        .slice(path.indexOf('related_content') + 1)
        .join('/') string;
    } else {
      file = path.join('/') string;
    }
```
just a couple of changes. Filepath to avoid confusion with file object, and also join returns string so shouldn't need the as. My understanding of 'as' and 'any' is to use it a little as possible 😅

Reply:
although I just realised 'file' is used as part of the interface ContentItem so perhaps might not be worth the change...